### PR TITLE
Add Rspec test for Tic-Tac-Toe

### PR DESCRIPTION
**Pull request by Yors-git**
In this branch I worked on the following:
- Init RSpec on the project folder
- Created a test for the display_board method
- Created 3 tests for the valid_input method
- Created a test for the wining_cases method

### CODE REVIEW

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General 
* [x] Let test the `Player` class
- [x] Each  spec should cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))
### [OPTIONAL]
- [ ] We favour the division of code in many methods. (This makes debugging easier and the code more readable). This applies to functions too.

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: spec/tic_tac_toe_game_logic_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  describe '#winning_cases' do
+    it 'checks the case of XXX' do
+      win_case = GameLogic.new
+      win_case.board[3] = 'X'
+      win_case.board[4] = 'X'
+      win_case.board[5] = 'X'
+      puts win_case.display_board
+      puts win_case.winning_cases
+      expect(win_case.winning_cases).to eql('X')
+    end
```
Comment (kobiyoyo) [inline spec/tic_tac_toe_game_logic_spec.rb]: - [x] You should properly test for a win case ,take note of the cells been used.

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General 
- [x] Test for when there is  a tie.
- [x] Check the comment below

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: spec/tic_tac_toe_game_logic_spec.rb (hunk 1 above)
Comment (Yors-git) [reply spec/tic_tac_toe_game_logic_spec.rb]: I reached to kobiyoyo and explained that the cells start from 0 in the test cases

Comment (kobiyoyo) [reply spec/tic_tac_toe_game_logic_spec.rb]: Please ignore this requirement, as it has been solved already.

**Review by acushlakoncept — APPROVED** [review]
# Status: PROJECT APPROVED  🎉 🚀

### Great Job Team 👍 

Excellent job on your project so far, well done on implementing all the requested changes 👏 

I will go ahead and mark this as complete, feel free to merge when you are ready. But before that, 

- [x] remove  

1 | 2 | 3 
-----------
 O | X | X 
-----------
 7 | 8 | 9 

from the output of your test results.

- [x] [OPTIONAL] You could use `let(:win_case) { 'GameLogic.new' } ` same goes with `player = Player.new` to avoid repeating yourself in all describe blockss.

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Pull request merged**